### PR TITLE
Tickets/dm 26101

### PR DESCRIPTION
**Pull request by iagaponenko**

### CODE REVIEW

File: core/modules/replica/DatabaseServices.cc [hunk 1]
```diff
@@ -167,6 +167,25 @@ json TransactionInfo::toJson() const {
...
 
+json TransactionContribInfo::toJson() const {
+    json info;
+    info["id"] = id;
+    info["transaction_id"] = transactionId;
+    info["worker"]     = worker;
+    info["database"]   = database;
+    info["table"]      = table;
+    info["chunk"]      = chunk;
+    info["overlap"]    = isOverlap ? 1 : 0;
```
Comment (n8pease) [inline core/modules/replica/DatabaseServices.cc:178]: just curious, where does `isOverlap` come from? I don't see it anywhere else in this file.

File: core/modules/replica/DatabaseServicesMySQL.h [hunk 2]
```diff
@@ -348,6 +366,12 @@ class DatabaseServicesMySQL : public DatabaseServices {
     std::vector<TransactionInfo> _findTransactionsImpl(util::Lock const& lock,
                                                        std::string const& predicate);
 
+    TransactionContribInfo _transactionContribImpl(util::Lock const& lock,
+                                                   std::string const& predicate);
+
+    std::vector<TransactionContribInfo> _transactionContribsImpl(util::Lock const& lock,
+                                                                 std::string const& predicate);
```
Comment (n8pease) [inline core/modules/replica/DatabaseServicesMySQL.h:373]: are these Impl functions still needed? I can't tell that they're being called anywhere.

File: core/modules/replica/DatabaseServices.cc (hunk 1 above)
Comment (iagaponenko) [reply core/modules/replica/DatabaseServices.cc:178]: This is a data member of the class (see the class's header `core/modules/replica/DatabaseServices.h`):
```
class TransactionContribInfo {
public:
    ...
    bool isOverlap = false;     /// (optional) A flavor of the chunked table (partitioned tables only)
```

File: core/modules/replica/DatabaseServicesMySQL.h (hunk 2 above)
Comment (iagaponenko) [reply core/modules/replica/DatabaseServicesMySQL.h:373]: The implementation is in the corresponding `CC` file `core/modules/replica/DatabaseServicesMySQL.cc`:
```
vector<TransactionContribInfo> DatabaseServicesMySQL::_transactionContribsImpl(
        util::Lock const& lock,
        string const& predicate) {

    string const context = _context(__func__) + "predicate=" + predicate + " ";
    ...
```

File: core/modules/replica/DatabaseServices.cc (hunk 1 above)
Comment (n8pease) [reply core/modules/replica/DatabaseServices.cc:178]: lol, thx. I've got 🐍-brain I guess.

File: core/modules/replica/DatabaseServicesMySQL.h (hunk 2 above)
Comment (n8pease) [reply core/modules/replica/DatabaseServicesMySQL.h:373]: huh, I thought I looked there. Ok, thx.

**Pull request merged**